### ISSUE TITLE
Unable to connect to database: SequelizeConnectionRefusedError: connect ECONNREFUSED 127.0.0.1:3306

### ISSUE THREAD

**Issue by bufanda** [linuxserver/docker-codimd#7]
I've setup my server with using your compose example. But I get the titled error. 
What I don't get, I have set the maraidb-container as DB-Host, in my configuration it Should be then something like `172.21.0.3` and not the local host. I even set the same configuration in my coded config.json, but it still has the same error. Is this a codimd problem in their latest release? The Database-Container is working an I can login from the Host-Machine.

**Comment by tobbenb** [comment]
Post your compose setup

**Comment by bufanda** [comment]
My compose is basically copied from the hub-page, I just added a reverse proxy.
```
version: "2"

networks:
  int.codimd:

services:
  mariadb:
    image: linuxserver/mariadb:latest
    container_name: codimd_mariadb
    restart: always
    volumes:
      - /volumes/codimd/db:/config
    environment:
      - MYSQL_ROOT_PASSWORD=password
      - MYSQL_DATABASE=codimd
      - MYSQL_USER=codimd
      - MYSQL_PASSWORD=password
      - PGID=1000
      - PUID=1000
      - TZ=Europe/London
    networks:
      int.codimd:

  codimd:
    image: linuxserver/codimd:1.4.0-ls8
    container_name: codimd
    restart: always
    depends_on:
      - mariadb
    volumes:
      - /volumes/codimd/app:/config
    environment:
      - DB_HOST=mariadb
      - DB_USER=codimd
      - DB_PASS=password
      - DB_NAME=coded
      - DB_PORT=3306
      - PGID=1000
      - PUID=1000
      - TZ=Europe/London
    networks:
      int.codimd:


  proxy:
    image: nginx:latest
    depends_on: 
      - codimd
    volumes:
      - /volumes/codimd/proxy/nginx.conf:/etc/nginx/nginx.conf:ro
      - /etc/letsencrypt:/etc/letsencrypt:ro
      - /etc/ssl/certs/dhparam.pem:/etc/nginx/dhparam.pem:ro
      - /volumes/codimd/proxy/logs:/etc/nginx/logs
    ports:
      - "3000:3000"
    networks:
      int.codimd:
```

My config.json
```
{
    "production": {
        "domain": "my.domain.com",
        "loglevel": "info",
        "db": {
            "username": "codimd",
            "password": "password",
            "database": "codimd",
            "host": "mariadb",
            "port": "3306",
            "dialect": "mysql"
        },
        "hsts": {
            "enable": true,
            "maxAgeSeconds": 31536000,
            "includeSubdomains": true,
            "preload": true
        },
        "csp": {
            "enable": true,
            "directives": {
            },
            "upgradeInsecureRequests": "auto",
            "addDefaults": true,
            "addDisqus": false,
            "addGoogleAnalytics": false
        },
        "mattermost": {
            "baseURL": "https://mymattermost/channel",
            "clientID": "....",
            "clientSecret": "...."
        }
    }
}
```

**Comment by aptalca** [comment]
Your mysql settings don't match.
1) for codimd you set the mysql host address to mariadb, but it should be codimd_mariadb, the name of the mariadb container
2) for codimd you set the database name to coded but it should be codimd, which is the database name you set in the mariadb environment variables

**Comment by bufanda** [comment]
> Your mysql settings don't match.
> 
> 1. for codimd you set the mysql host address to mariadb, but it should be codimd_mariadb, the name of the mariadb container
I don't believe so, I copied the original compose file from the readme to a second host in my homelab and it worked.
> 2. for codimd you set the database name to coded but it should be codimd, which is the database name you set in the mariadb environment variables
Right but this wasn't the root cause of the problem.
The Problem occurred when I started to use my config.json. Even though I added the db connection infos it still tried to connect to localhost. I had to tell in the config to use the environment variables for DB connection. Then I had the problem that the MariaDB-Container wasn't running on Ubuntu 14.04 anymore, so I switched to Postgres as Database, now it works.

But anyway. Time to update the Host :)
